rpi_backlight.py

```diff
@@ -38,97 +39,174 @@ def _get_value(name: str) -> str:
+    if mode == "MODE_TINKERBOARD":
+        _set_value("tinker_mcu_bl", value)
+    elif mode == "MODE_RPI":
+        _set_value("bl_power", value)
 
+def get_brightness_value() -> int:
+    """Return the actual display brightness."""
+    if mode == "MODE_TINKERBOARD":
+        return int(_get_value("tinker_mcu_bl"))
+    elif mode == "MODE_RPI":
+        return int(_get_value("actual_brightness"))
 
 def get_max_brightness() -> int:
     """Return the maximum display brightness."""
-    return int(_get_value("max_brightness"))
-
+    if mode == "MODE_TINKERBOARD":
+        return 255
+    elif mode == "MODE_RPI":
+        return int(_get_value("max_brightness"))
+    
 
 def get_power() -> bool:
-    """Return wether the display is powered on or not."""
+    """Return whether the display is powered on or not."""
     # 0 is on, 1 is off
-    return not int(_get_value("bl_power"))
+    if mode == "MODE_TINKERBOARD":
+        return not int(_get_value("tinker_mcu_bl"))
+    elif mode == "MODE_RPI":
+        return not int(_get_value("bl_power"))
 
 
-def set_brightness(value: int, smooth: bool = False, duration: float = 1) -> None:
+def set_brightness(value: int, smooth: bool = True, duration: float = 1) -> None:
     """Set the display brightness.
 
-    :param value: Brightness value between 11 and 255
+    :param value: Brightness value between 0 and 255
     :param smooth: Boolean if the brightness should be faded or not
     :param duration: Fading duration in seconds
     """
-
     max_value = get_max_brightness()
     if not isinstance(value, int):
         raise ValueError("integer required, got '{}'".format(type(value)))
-    if not 10 < value <= max_value:
-        raise ValueError(
-            "value must be between 11 and {}, got {}".format(max_value, value)
-        )
+
+    if mode == "MODE_TINKERBOARD":
+            if not -1 < value <= max_value:
+                raise ValueError(
+                    "value must be between 0 and {}, got {}".format(max_value, value)
+                )
+    elif mode == "MODE_RPI":
+        if not -10 < value <= max_value:
+                raise ValueError(
+                    "value must be between 0 and {}, got {}".format(max_value, value)
+                )
 
     if smooth:
         if not isinstance(duration, (int, float)):
             raise ValueError(
                 "integer or float required, got '{}'".format(type(duration))
             )
-        actual = get_actual_brightness()
+        actual = get_brightness_value()
         diff = abs(value - actual)
         while actual != value:
             actual = actual - 1 if actual > value else actual + 1
-
-            _set_value("brightness", actual)
+            set_brightness_value(actual)
             time.sleep(duration / diff)
     else:
-        _set_value("brightness", value)
+        set_brightness_value(value)
 
 
-def set_power(on: bool) -> None:
+def set_power(on: bool, smooth: bool = True, duration: float = 1) -> None:
```

Comment:
Please remove these arguments, setting the power on or of implies a binary state and should not have a `smooth` parameter (it wouldn't even work for Raspberry Pi users). If people want to turn down the brightness smoothly and turn the display off afterwards, they have two specific functions to do so.

Reply:
With the defaults set, this should not break anything, but rather, offer additional functionality for those who wish to use it.
It is set to false (so when you run it it should not do anything different)

Reply:
Keep in mind, setting the power for those who use Tinkerboard means actually dimming brightness to 0. So powering off smoothly makes sense for those who want it

Reply:
Sure, but there's `set_brightness` for that.

Reply:
But as a library, I want people to be able to still set the power, but it only offers them the ability to do it smoothly. But because there are defaults nothing is broken.
I will think philosophically today for which makes sense

Reply:
Also, it is still binary on or off. It just gives option to make it much more appealing to --toggle -s (toggle smoothly)
Or --on -s (power on smoothly)

Reply:
Sorry, but I disagree - on and off is on and off, any transition between should not be possible. That's setting the brightness, and that's what `set_brightness` is for. FWIW it violates the Zen of Python:

> There should be one-- and preferably only one --obvious way to do it.

Reply:
Fixed